src/components/Swap/Input.tsx

```diff
@@ -52,6 +52,35 @@ export const InputColumn = styled(Column)<{ approved?: boolean; hasColor?: boole
     filter: ${({ approved }) => (approved ? undefined : 'saturate(0) opacity(0.4)')};
     transition: filter 0.25s;
   }
+
+  &:before {
+    box-sizing: border-box;
+    background-size: 100%;
+    border-radius: inherit;
+
+    position: absolute;
+    top: 0;
+    left: 0;
+
+    width: 100%;
+    height: 100%;
+    pointer-events: none;
+    content: '';
+    border: 1px solid ${({ theme }) => theme.module};
+    transition: 125ms ease border-color;
+  }
+
+  ${({ theme, disableHover }) =>
```

Comment:
nit: these two could be put into one conditional.
another nit: I find this easier to read if the conditional is within border-color, and not over a larger css block.